middleware/middleware_permission.go

```diff
@@ -1,18 +1,26 @@
 package middleware
 
 import (
+	"log/slog"
 	"net/http"
 
-	"github.com/0xsequence/authcontrol"
 	"github.com/0xsequence/quotacontrol/proto"
+	"github.com/goware/logger"
 )
 
 // EnsurePermission middleware that checks if the session type has the required permission.
-func EnsurePermission(client Client, minPermission proto.UserPermission, eh authcontrol.ErrHandler) func(next http.Handler) http.Handler {
-	if eh == nil {
-		eh = errHandler
+func EnsurePermission(client Client, minPermission proto.UserPermission, o *Options) func(next http.Handler) http.Handler {
+	eh := errHandler
+	if o != nil && o.ErrHandler != nil {
+		eh = o.ErrHandler
 	}
 
+	logger := logger.NewLogger(logger.LogLevel_DEBUG)
+	if o != nil && o.Logger != nil {
+		logger = o.Logger
+	}
+	logger = logger.With(slog.String("middleware", "ratelimit"))
```

Comment:
```suggestion
	logger = logger.With(slog.String("middleware", "ensurePermission"))
```
Here is already same static log property:value https://github.com/0xsequence/quotacontrol/pull/57/files#diff-89099739e0265112baf868728d4eb7b062297a5e722c7d55e43f51626987ca66R69